app/services/host_inventory_api.rb

```diff
@@ -45,7 +45,7 @@ def inventory_host
   def find_results(body)
     body['results'].find do |host|
       host['account'] == @account.account_number && (
-        host['id'] == @host_inventory_id ||
+        host['id'] == @id ||
```

Comment:
This was never correct (`@host_inventory_id` was always `nil`). We've been matching on hostname this whole time for report parsing.